src/trade-params/saga/DurationSaga.js

```diff
@@ -142,7 +142,9 @@ export function* handleStartTimeChange(action) {
     const contractPerType = contractNeeded[tradeCategory][type];
     const durationAllowed = allTimeRelatedFieldValid(newDateStart, duration, durationUnit, contractPerType);
 
-    if (durationAllowed) {
+     if (isValidTime(time)) {
+        yield put(updateTradeError(index, 'durationError', 'Time format is wrong!'));
```

Comment:
I think either of the two serve our purpose here, There isn't a code block after the if statement so the block exit there. Plus 'return' is part of what the yield keyword does it self if i understand correctly. I think either of the two should be fine(i.e return or not ). https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/yield